.github/workflows/test-and-release.yml

```diff
@@ -30,3 +30,17 @@ jobs:
           githubToken: ${{ secrets.GITHUB_TOKEN  }}
           dockerUsername: ${{ secrets.GITLEAKS_DOCKER_USERNAME }}
           dockerPassword: ${{ secrets.GITLEAKS_DOCKER_PASSWORD }}
+  release:
+    needs: [build, test]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Semantic Release
+        id: semantic
+        uses: codfish/semantic-release-action@4e9fa8ec064813465dfeccac8ae70f1348fb5dec
```

Comment:
This is one approach, but also there is this alternative https://github.com/Typeform/pharos/blob/master/.github/workflows/ci.yml#L59 which I really recommend, because you can execute locally for testing purposes (it does not release from local)